runtime/pubsub/test_internal.go

```diff
@@ -0,0 +1,15 @@
+package pubsub
+
+import (
+	"encore.dev/pubsub/internal/test"
+	"encore.dev/runtime"
+)
+
+func GetTestTopicInstance[T any](topic *Topic[T]) any {
+	testTopic, ok := topic.topic.(*test.TestTopic[T])
+	if !ok {
+		panic("testTopic not called with a test topic")
+	}
+
+	return testTopic.TestInstance(runtime.CurrentTest())
+}
```

Comment:
This isn't a public API (hence the `_internal` suffix on the file).

This is the linking piece between the `pubsub` unexported stuff, and the new `test` package